docs/DeveloperGuide.md

```diff
@@ -1,33 +1,373 @@
 # Developer Guide
 
+* Table of Contents
+{:toc}
+
+## Introduction
+Welcome to Study It Developer Guide!
+
+This document details the architecture of Study It. It aims to provide an overview of the high-level system 
+architecture and design of the app. It will also break down the design of the app into smaller components and be 
+explained in detail. This will allow the readers to understand the architecture and design flow of Study It and 
+how it operates.
+
+Study It is an interactive desktop app that helps NUS students manage their study related matters especially during 
+this online study environment. It is optimized for use via a Command Line Interface (CLI). It can keep track of the 
+student’s various study matters, present them in an organized and thoughtful manner to help them organize their 
+study life. It will also provide various functionalities to help with their studies. 
+
+The 4 main functionalities available now are bookmark, timetable, academic tracker and flashcard. 
+Each of this will be discussed in detail under the “Design & Implementation” section later in the document.
+
+
+## Setting up & getting started
+**Requirements**
+* Java 11.0.8
+  * You may refer to this [website](https://www.oracle.com/java/technologies/javase/jdk11-archive-downloads.html) 
+to download and install Java 11.0.8
+  * The installation guide can be found in the following websites:
+    * [Windows](https://docs.oracle.com/en/java/javase/11/install/installation-jdk-microsoft-windows-platforms.html#GUID-A7E27B90-A28D-4237-9383-A58B416071CA)
+    * [Linux](https://docs.oracle.com/en/java/javase/11/install/installation-jdk-linux-platforms.html)
+    * [Mac](https://docs.oracle.com/javase/10/install/installation-jdk-and-jre-macos.htm#JSJIG-GUID-2FE451B0-9572-4E38-A1A5-568B77B146DE)
+
+**Running the software**
+1. Download the tP.jar file from our [Github release](https://github.com/AY2021S1-CS2113T-T12-1/tp/releases) and 
+place it in an empty folder
+2. Open your computer’s command prompt
+3. Change the directory of the command prompt to the folder containing tP.jar file
+4. Type `java -jar tp.jar` into the command prompt and press Enter to execute it
+5. If the application runs successfully, you’ll be greeted by a welcome message
+
+**Setting up the project on your PC**
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. Configure the JDK: Configuring the JDK to ensure Intellij is configured to use JDK 11.
+2. Import the project as a Gradle project.
+3. Run the studyit.StudyIt and try a few commands.
+4. Run the tests to ensure they all pass.
+
 ## Design & implementation
+This section explains the architecture of our software and the design of each component.
+
+### **Major components**
+This section introduces the multiple components in Study It.
+
+![](Images/ArchitectureDiagrams/components.png)
```

Comment:
Perhaps can remove the small negative sign box on the top of every class box. It can be a little bit confusing. 
![image](https://user-images.githubusercontent.com/60136318/97657059-e7730b80-1aa3-11eb-9df8-be00a5e22d8d.png)